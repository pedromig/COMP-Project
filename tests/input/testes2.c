int f1(void, int a){
    return 0;
}

void f2(void){
    //algo
}

int a;

int main(void){


    if(a == f2()){ }

    if(a == f2(33)){ }

}
