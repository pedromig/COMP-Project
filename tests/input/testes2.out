Line 1, col 8: Invalid use of void type in declaration
Line 14, col 10: Operator == cannot be applied to types int, void
Line 16, col 13: Wrong number of arguments to function f2 (got 1, required 0)
Line 16, col 10: Operator == cannot be applied to types int, void
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
f2	void(void)
a	int
main	int(void)

===== Function f2 Symbol Table =====
return	void

===== Function main Symbol Table =====
return	int

Program
..FuncDefinition
....Int
....Id(f1)
....ParamList
......ParamDeclaration
........Void
......ParamDeclaration
........Int
........Id(a)
....FuncBody
......Return
........IntLit(0)
..FuncDefinition
....Void
....Id(f2)
....ParamList
......ParamDeclaration
........Void
....FuncBody
..Declaration
....Int
....Id(a)
..FuncDefinition
....Int
....Id(main)
....ParamList
......ParamDeclaration
........Void
....FuncBody
......If
........Eq - int
..........Id(a) - int
..........Call - void
............Id(f2) - void(void)
........Null
........Null
......If
........Eq - int
..........Id(a) - int
..........Call - void
............Id(f2) - void(void)
............IntLit(33) - int
........Null
........Null
